pyani/scripts/pyani_script.py

```diff
@@ -103,6 +103,10 @@ def run_main(argv: Optional[List[str]] = None) -> int:
         sys.stderr.write(f"{VERSION_INFO}\n")
         return 0
 
+    # If the command run is not pyani (e.g., `pytest`, then we
+    # don't want to apply pyani-specific checks)
+    if len(sys.argv) == 1 and not sys.argv[0].endswith("pyani"):
```

Comment:
I don't understand why `endswith()` is used in preference to `==`. What is your thinking, here?

Reply:
Because `sys.argv[0]` in this case is not `pyani`, but:

`/Users/baileythegreen/Software/miniconda3/envs/pyani_dev/bin/pyani`.

The same goes for `pytest` above. I tried `==` first, and it failed.

Reply:
I see. It would be more robust to either (i) treat `sys.argv[0]` as a `Path` object and using the `.name` attribute, or (ii) `.split("/")` the passed path and test the last element in the resulting collection. Using `.endswith()` does give a `True` value with the case you want, but also in cases where the executable is not simply `pyani` (or `pytest`, in the case where you'r echecking for that):

```python
>>> from pathlib import Path
>>> pyani_path = "/my/path/to/pyani"
>>> notpyani_path = "/my/path/to/notpyani"
>>> Path(pyani_path).name
'pyani'
>>> Path(notpyani_path).name
'notpyani'
>>> pyani_path.split("/")[-1]
'pyani'
>>> notpyani_path.split("/")[-1]
'notpyani'
>>> pyani_path.endswith("pyani")
True
>>> notpyani_path.endswith("pyani")
True
```

Reply:
I mentioned in our last meeting that it had occurred to me that I'd thought of a better way to do it, and I *think* I said that was using a `Path()` object; you said you thought not testing for `pytest` specifically would be better—and I thought part of the implication there was you didn't want `Path()` used.

I must have misunderstood; I can of course use `Path()` and test for whether the program is `pyani`, or not.